game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -689,10 +690,18 @@ private void setCanNotTarget(final Set<UnitType> value) {
 
   public Set<UnitType> getCanNotTarget() {
     if (canNotTarget == null) {
-      canNotTarget =
-          new HashSet<>(
-              CollectionUtils.getMatches(
-                  getData().getUnitTypeList().getAllUnitTypes(), Matches.unitTypeIsAir()));
+      if (isSuicideOnAttack && isFirstStrike) { // Support isSuicide shortcut
+        canNotTarget =
+            new HashSet<>(
+                CollectionUtils.getMatches(
+                    getData().getUnitTypeList().getAllUnitTypes(),
+                    Matches.unitTypeIsSuicideOnAttack().or(Matches.unitTypeIsSuicideOnDefense())));
```

Comment:
nit, it's just the matches type that varies depending on the if condition, seems though the if/else could be simplified.

Beyond the code low-level cleanup, why is the 'isSub' and 'isSuicide' properties opposites of one another? IE: why is if 'isSuicide' is false, then 'isSub' is true?

Reply:
Possibly though I was concerned with it being harder to read if too much of it was inlined.

They aren't true opposites but because this is for supporting the deprecated properties, current maps never have isSub and isSuicide on the same unit as they didn't work together.

Reply:
I don't think that would qualify as too much inlining, it helps a lot as it makes clear what is different between the two cases. You can assign to an intermediary variable if the inline syntax feels awkward.

> They aren't true opposites...

Point taken, a casual reader would be very confused why we can assume with one condition being false that we fall into this other (seemingly very unrelated) case.

Reply:
Updated.